internal/fs/inode/dir.go

```diff
@@ -836,9 +836,11 @@ func (d *dirInode) DeleteChildDir(
 
 	// if the directory is an implicit directory, then no backing object
 	// exists in the gcs bucket, so returning from here.
-	if isImplicitDir {
```

Comment:
This method should be refactored and isImplicitDir check should be removed, delete method's behaviour should only be for delete, it should not have intelligence to know if passed directory is implicit or not, the caller passing isImplicitDir check is already aware that given directory is implicit or not, it should not call this method in case implicit dir is true.

Reply:
As discussed ### offline, we considered implementing this logic outside the existing structure. However, we realized that this would necessitate a hierarchical bucket check due to the lack of implicit directories in such buckets. This check would need to be duplicated in various places, leading us to abandon this approach.